x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/parser/AbstractBuilder.java

```diff
@@ -183,6 +184,48 @@ public static String unquoteString(Source source) {
         return resultString.toString();
     }
 
+    private static String handleUnicodeChars(Source source, String text) {
+        StringBuilder sb = new StringBuilder();
+
+        int startIdx = 0;
+        int endIdx = 0;
+        int idx = text.indexOf("\\u");
+        while (idx >= 0) {
+            String unicodeSequence;
+            if (text.charAt(idx + 2) == '{') {
+                endIdx = text.indexOf("}", idx + 1) + 1;
+                unicodeSequence = text.substring(idx + 3, endIdx - 1);
+                int length = unicodeSequence.length();
+                if (length < 2 || length > 8) {
+                    throw new ParsingException(source, "Unicode sequence in curly braces should use [2-8] hex digits, [{}] has [{}]",
+                            text.substring(idx, endIdx), length);
+                }
+            } else {
+                endIdx = idx + 6;
+                unicodeSequence = text.substring(idx + 2, endIdx);
+            }
+            sb.append(text, startIdx, idx).append(hexToUnicode(source, unicodeSequence));
+            idx = text.indexOf("\\u", endIdx);
+            startIdx = endIdx;
+        }
+        if (endIdx < text.length()) {
+            sb.append(text.substring(endIdx));
+        }
+        return sb.toString();
+    }
+
+    private static String hexToUnicode(Source source, String hex) {
+        int code = Integer.parseInt(hex, 16);
+        if (code == 0) {
+            throw new ParsingException(source, "Unicode sequence results in null");
```

Comment:
Why is `\0` prohibited? Couldn't this be an issue, with ES accepting `\u00`?

Reply:
Not sure about that, followed suggestion by @rw-access, Is there a reason that we don't want the null?

Reply:
feel free to keep `\0` here if you like.
I had to drop it on my end, because C strings are null terminated and I wanted to avoid downstream issues